File: Models/Zadruga.cs
namespace Models;

public class Zadruga {

    [Key]
    public int ID { get; set; } 
    [Required]
    [MaxLength(20)]
    public string UserName { get; set; }

    [Required]
    [MaxLength(20)]
    [RegularExpression(@"^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$",ErrorMessage = "Nevalidan password!")]
    public string Password { get; set; }

    [Required]
    [MinLength(9)]
    [MaxLength(9)]
    [RegularExpression(@"\d+")]
    public string PIB { get; set; }

    [Required]
    [MinLength(8)]
    [MaxLength(8)]
    [RegularExpression(@"\d+")]
    public string MaticniBroj { get; set; }

    [MaxLength(300)]
    public string Slika { get; set; }

    [MaxLength(100)]
    [DataType(DataType.EmailAddress)]
    [EmailAddress]
    public string Email { get; set; }

    [MaxLength(20)]
    [RegularExpression(@"\d+")]
    public string FiksniTelefon { get; set; }

    [MaxLength(20)]
    [RegularExpression(@"\d+")]
    public string MobilniTelefon { get; set; }

    [Required]
    [MaxLength(50)]
    public string Naziv { get; set; }

    [Required]
    [MaxLength(30)]
    [RegularExpression(@"\d+")]
    public string BrojRacuna { get; set; }

    [Required]
    [MaxLength(20)]
    public string Grad { get; set; }

    [Required]
    [MaxLength(50)]
    public string Ulica { get; set; }

    [MaxLength(3)]
    public string BrojStana { get; set; }

    [MaxLength(3)]
    public string BrojUlaza { get; set; }

    [MaxLength(10000)]
    public string Informacije { get; set; }
    
    [JsonIgnore]
    public List<Feedback> Feedbacks { get; set; }
    [JsonIgnore]
    public List<Oglas> Oglasi { get; set; }
    [JsonIgnore]
    public List<Ugovor> Ugovori { get; set; }
    [JsonIgnore]
    public List<NotifikacijaZadruga> NotifikacijeZadruge { get; set; }

    
}

File: Client/Zadruga/JS/Zadruga.js
export class Zadruga{
    constructor(id, username, password, pib, maticniBroj, slika, email, fiksniTelefon,
        mobilniTelefon, naziv, brojRacuna, 
        grad, ulica, brojStana, brojUlaza, informacije,feedbacks, oglasi, ugovori, notifikacije){
            this.id=id;
            this.username=username;
            this.password=password;
            this.pib=pib;
            this.maticniBroj=maticniBroj;
            this.slika=slika;
            this.email=email;
            this.fiksniTelefon=fiksniTelefon;
            this.mobilniTelefon=mobilniTelefon;
            this.naziv=naziv;
            this.brojRacuna=brojRacuna;
            this.grad=grad;
            this.ulica=ulica;
            this.brojStana=brojStana;
            this.brojUlaza=brojUlaza;
            this.informacije=informacije;
            this.feedbacks=feedbacks;
            this.oglasi=oglasi;
            this.ugovori=ugovori;
            this.notifikacije=notifikacije;
    }

    crtajZadruguAdmin(host){
        let zadrugaDiv = document.createElement("div");
        zadrugaDiv.classList.add("zadrugaDiv");
        if(this.email==="izbrisan korisnik")
            zadrugaDiv.classList.add("izbrisanKorisnik");
        host.appendChild(zadrugaDiv);

        //slika + srednjiDiv
        let slikaSrednjiDiv = document.createElement("div");
        slikaSrednjiDiv.classList.add("slikaSrednjiDiv");
        zadrugaDiv.appendChild(slikaSrednjiDiv);

        //slika
        let slikaDiv = document.createElement("div");
        slikaDiv.classList.add("slikaDiv");
        slikaSrednjiDiv.appendChild(slikaDiv);

        let zadrugaSlika = document.createElement("img");
        zadrugaSlika.classList.add("zadrugaSlika");
        zadrugaSlika.src=`../../Slike/`+this.slika;
        slikaDiv.appendChild(zadrugaSlika);

        //ime i prezime, grad, godine, ocena
        let srednjiDiv = document.createElement("div");
        srednjiDiv.classList.add("srednjiDiv");
        slikaSrednjiDiv.appendChild(srednjiDiv);

        //ime i prezime
        let nazivDiv = document.createElement("div");
        nazivDiv.classList.add("nazivDiv","zadruga","imePrezimeDiv");
        nazivDiv.id=this.username;
        srednjiDiv.appendChild(nazivDiv);

        let nazivVrednost = document.createElement("label");
        nazivVrednost.classList.add("imePrezimeVrednost");
        nazivVrednost.innerHTML=`Zadruga: ${this.naziv}`;
        nazivDiv.appendChild(nazivVrednost);

        //grad, godine
        let zGradGodineDiv = document.createElement("div");
        zGradGodineDiv.classList.add("zGradGodineDiv");
        srednjiDiv.appendChild(zGradGodineDiv);

        //grad
        let zGradDiv = document.createElement("div");
        zGradDiv.classList.add("zGradDiv");
        zGradGodineDiv.appendChild(zGradDiv);

        let zGradVrednost = document.createElement("label");
        zGradVrednost.classList.add("zGradVrednost");
        zGradVrednost.innerHTML=`Grad: ${this.grad}`;
        zGradDiv.appendChild(zGradVrednost);

        //godine
        let godineDiv = document.createElement("div");
        godineDiv.classList.add("godineDiv");
        zGradGodineDiv.appendChild(godineDiv);

        let godineVrednost = document.createElement("label");
        godineVrednost.classList.add("godineVrednost");
        godineVrednost.innerHTML = `Kontakt: ${this.mobilniTelefon}`;
        godineDiv.appendChild(godineVrednost);

        //ocena
        let ocenaDiv = document.createElement("div");
        ocenaDiv.classList.add("ocenaDiv");
        srednjiDiv.appendChild(ocenaDiv);

        let ocenaVrednost = document.createElement("label");
        ocenaVrednost.classList.add("ocenaVrednost");
        fetch(`http://localhost:5258/Feedback/VratiProsecnuOcenuZadruga/${this.id}`)
        .then(p=>p.json().then(prosecnaOcena=>{  
            ocenaVrednost.innerHTML = `Ocena: ${prosecnaOcena}`;
        }));
        ocenaDiv.appendChild(ocenaVrednost);

        //odabran (checkbox)
        let zadrugaDugmadDiv = document.createElement("div");
        zadrugaDugmadDiv.classList.add("zadrugaDugmadDiv");
        zadrugaDiv.appendChild(zadrugaDugmadDiv);

        if(this.email!=="izbrisan korisnik")
        {
            let obrisiZadruguBtn = document.createElement("button");
            obrisiZadruguBtn.classList.add("obrisiZadruguBtn");
            obrisiZadruguBtn.classList.add("obrisiKorisnika");
            obrisiZadruguBtn.id=this.username;
            obrisiZadruguBtn.innerHTML="Obriši korisnika";
            zadrugaDugmadDiv.appendChild(obrisiZadruguBtn);
        }
        
        let otvoriZadruguBtn = document.createElement("button");
        otvoriZadruguBtn.classList.add("otvoriZadruguBtn");
        otvoriZadruguBtn.classList.add("otvoriKorisnika");
        otvoriZadruguBtn.id=this.username;
        otvoriZadruguBtn.innerHTML="Prikaži korisnika";
        zadrugaDugmadDiv.appendChild(otvoriZadruguBtn);
    }

}